plugins/dap/dap-ui.nix

```diff
@@ -0,0 +1,209 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.dap.extensions.dap-ui;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption =
+    helpers.mkNullOrOption
+    (
+      with types;
+        either int (numbers.between 0.0 1.0)
+    );
+
+  mkMappingOption =
+    helpers.defaultNullOpts.mkNullable
+    (
+      with types;
+        attrsOf (either str (listOf str))
+    );
+
+  layoutOption = types.submodule {
+    options = {
+      elements = mkOption {
+        default = [];
+        description = "Elements to display in this layout.";
+        type = types.listOf types.attrs;
+      };
+
+      size = mkOption {
+        default = 10;
+        description = "Size of the layout in lines/columns.";
+        type = types.int;
+      };
+
+      position = mkOption {
+        default = "left";
+        description = "Which side of editor to open layout on.";
+        type = types.enum ["left" "right" "top" "bottom"];
+      };
+
+      id = helpers.mkNullOrOption types.str "Element ID.";
```

Comment:
I don't see `id` as a top level option

Reply:
https://github.com/rcarriga/nvim-dap-ui/blob/c020f660b02772f9f3d11f599fefad3268628a9e/doc/nvim-dap-ui.txt#L286

Reply:
```
---@class dapui.Config.layout
---@field elements string[]|dapui.Config.layout.element[] Elements to display
--- in this layout
---@field size number Size of the layout in lines/columns
---@field position "left"|"right"|"top"|"bottom" Which side of editor to open
--- layout on
```

Well I guess either the comments or the documentation are misleading, let's follow the documentation

Reply:
It may be the dapui.Config.layout documentation that is not separated from the dapui.Config.layout.element documentation though.

Reply:
Taking a closer look at the source file, I think that this is the case, should we remove it then?